### DISCUSSION TITLE
[Playwright] Save video even for tests that passed and resolution

### DISCUSSION THREAD

**Discussion by incoming-th** [codeceptjs/CodeceptJS#4303] [Q&A]
Hi,

I am using Playwright helper and I have enabled the video. However it seems the video are saved in the output directory only when the tests failed. Is there a way to save them even if the tests succeeded?

Additionally, how to change the resolution of the video? I see there is an empty space for the subtitles, but when now using the subtitles this should not appear?

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/f2876abf-e9a7-4f45-926e-786a07cb6574)

Thanks

**Comment by incoming-th** [comment]
Answering my first question after looking at the code, need to add "keepVideoForPassedTests" to true:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/cbf71afe-1379-47c1-8e14-d7fb1f76f4a6)

**Comment by incoming-th** [comment]
For the video resolution, I have checked the Playwright option:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/585139a0-d816-4363-b710-a6fc45f81e32)

But it seems to never be called in the Playwright helper and use windowSize instead:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/6e04debb-0c8a-4eff-ae9c-08a4d02f1177)

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/4cb73947-15af-498d-ad26-69201bcf1938)

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/57817e13-bfb2-495b-b26e-3af4b47cace5)

Not sure if it's by design or that's not hwo to define the video resolution.

**Reply by kobenguyent** [reply]
shall be resolved with this https://github.com/codeceptjs/CodeceptJS/pull/4311

**Reply by incoming-th** [reply]
Excellent thanks a lot!
